app/src/main/java/com/example/criminalintentapp/presentation/authentication/RegisterActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.criminalintentapp.presentation.authentication
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import com.example.criminalintentapp.R
+import com.example.criminalintentapp.databinding.ActivityRegisterBinding
+import com.example.criminalintentapp.presentation.MainActivity
+
+class RegisterActivity : AppCompatActivity() {
+
+    private val authenticationViewModel: AuthenticationViewModel by lazy {
+        ViewModelProvider(this).get(AuthenticationViewModel::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = ActivityRegisterBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        initViewModelObservers()
+
+        setOnClickListeners(binding)
+    }
+
+    private fun initViewModelObservers() {
+        authenticationViewModel.userRegisterLiveData.observe(this) { user ->
+            user?.let {
+                Toast.makeText(this, "Registered successfully", Toast.LENGTH_SHORT).show()
```

Comment:
Hardcoded text, please change it to a string resource

Reply:
On a second thought, I think this will no longer be necessary when we implement the `automatic login` I was talking about on Teams. So please make the necessary adjustment so when user registers, he is then also logged in as this would be a far more better user experience than having to register and then fill the login fields again with the same credentials